notebook/static/tree/js/notebooklist.js

```diff
@@ -1434,6 +1434,7 @@ define([
         var upload_button = $('<button/>').text(i18n.msg._("Upload"))
             .addClass('btn btn-primary btn-xs upload_button')
             .click(function (e) {
+                item.find('.upload_button').text("Uploading...");
```

Comment:
This is the actual change. My JS is weak so if there is some more idiomatic way of doing this please show me :)